hbase-server/src/main/java/org/apache/hadoop/hbase/security/provider/SimpleSaslServerAuthenticationProvider.java

```
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.security.provider;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.security.sasl.SaslServer;
+
+import org.apache.hadoop.hbase.HBaseInterfaceAudience;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.token.SecretManager;
+import org.apache.hadoop.security.token.TokenIdentifier;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.apache.yetus.audience.InterfaceStability;
+
+@InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.AUTHENTICATION)
+@InterfaceStability.Evolving
+public class SimpleSaslServerAuthenticationProvider extends SimpleSaslClientAuthenticationProvider
+    implements SaslServerAuthenticationProvider {
+
+  @Override
+  public SaslServer createServer(SecretManager<TokenIdentifier> secretManager,
+      Map<String, String> saslProps) throws IOException {
+    return null;
```

Seems odd creating a null server.

---

Yeah, this is some of what I meant about `SIMPLE` being an oddball right now. SIMPLE auth'n in HBase doesn't go through SASL, so we have some code like this where we don't actually create some pieces. Long-term, I'd like to come back and push all auth'n through SASL (even SIMPLE, but not changing the semantics of what SIMPLE means to us).

---

Maybe we should rather throw an error here? My reading is that under `SIMPLE`, we should never create an `HBaseSaslRpcServer` instance, and even if we do, returning null here would just crash things further down in the code path. At least throwing an error here would explicitly show something is wrong.

---

Sure, that looks fine.